core/os/android/adb/screen.go

```diff
@@ -88,22 +79,22 @@ func (b *binding) UnlockScreen(ctx context.Context) (bool, error) {
 	case screenOnUnlocked:
 		return true, nil
 
-	case screenOffUnlocked:
-		if err := b.KeyEvent(ctx, android.KeyCode_Wakeup); err != nil {
-			return false, err
-		}
-		time.Sleep(keyEventEffectDelay)
-		return b.isScreenUnlocked(ctx)
-
-	case screenOffLocked:
+	default:
+		// Devices may do unexpected transitions between screen
+		// states, so this code does not try to be smart about
+		// expected state changes: unless screenOnUnlocked, apply the
+		// wakeup (put screen on) and menu (unlock screen if no
+		// credentials required) key event sequence.
 		if err := b.KeyEvent(ctx, android.KeyCode_Wakeup); err != nil {
 			return false, err
 		}
-		fallthrough
-	case screenOnLocked:
 		if err := b.KeyEvent(ctx, android.KeyCode_Menu); err != nil {
 			return false, err
 		}
+		// KeyEvent() eventually calls adb shell commands, which are
```

Comment:
This part of the comment is false. b.KeyEvent will only return once the adb command has finished. See the `Run()` function `core/os/shell/command.go`

Reply:
Yep, I got confused: adb shell commands are synchronous, some other commands are not (e.g. `adb am start` returns before the app is actually started). I updated the comment accordingly. Thanks!